conda_vendor/conda_vendor.py

```diff
@@ -33,38 +33,42 @@ def create_vendored_dir(environment_file, platform, desired_path=None) -> Path:
             sys.exit(f"Failed to read environment name from {environment_file}")
     
     # use current working directory if no path specified
+
+    def _create_vendored_dir(root_dir, env_name, platform):
+
+        if isinstance(root_dir, str):
+            root_dir = Path(root_dir) 
+
+        path = root_dir / env_name
+        os.mkdir(path)
+        create_platform_dir(path, platform)
+        create_noarch_dir(path)
+        return path
+
     if desired_path == None:
         try:
-            path = os.path.join(os.getcwd(), environment_name)
-            os.mkdir(path)
-            create_platform_dir(path, platform)
-            create_noarch_dir(path)
-            return path
+            return _create_vendored_dir(Path.cwd(), environment_name, platform)
         except FileExistsError as err:
             click.echo(err)
             sys.exit(f"Directory \"{environment_name}\" already exists")
     else:
         try:
-            path = os.path.join(desired_path, environment_name)
-            os.mkdir(path)
-            create_platform_dir(path, platform)
-            create_noarch_dir(path)
-            return path
+            return _create_vendored_dir(desired_path, environment_name, platform)
         except FileExistsError as err:
             click.echo(err)
             sys.exit(f"Directory \"{desired_path}/{environment_name}\" already exists")
 
 def create_platform_dir(path, platform):
```

Comment:
```suggestion
def create_platform_dir(path, platform, overwrite=True):
```
add option to automatically overwrite the existing directory if it exists, default to True.